codegenerator/cli/templates/static/codegen/src/globalState/GlobalStateManager.res

```diff
@@ -23,11 +23,11 @@ module MakeManager = (S: State) => {
         S.invalidatedActionReducer
       }
       let (nextState, nextTasks) = reducer(self.state, action)
-      self.state = nextState
       switch self.stateUpdatedHook {
-      | Some(hook) => hook(nextState)
-      | None => ()
+      | Some(hook) if self.state !== nextState => hook(nextState)
```

Comment:
Somewhat nuanced rescript maybe it deserves a comment
```suggestion
      // In ReScript `!==` is shallow equality check rather than `!=`
      // This is just a check to see if a new object reference was returned
      | Some(hook) if self.state !== nextState => hook(nextState)
```